### ISSUE TITLE
Icon paths not being followed correctly

### ISSUE THREAD

**Issue by Jemi** [lxde/lxqt#915]
Steam installs its icons to ~/.local/share/icons/hicolor/<res>/apps because I have ${HOME}/.local/share in my $XDG_DATA_DIRS. lxqt should look in these directories as per:

http://standards.freedesktop.org/icon-theme-spec/icon-theme-spec-latest.html

lxqt doesn't do thi, so this is a bug.

lxqt version: 0.10.0

sytem: Gentoo Linux

architecture: amd64

**Comment by palinek** [comment]
Could you, please, more precisely describe the scenario where icons are not found/shown? (which application etc..)

**Comment by Jemi** [comment]
Sorry, I meant to say that this is in reference to the shortcuts on the desktop. For example, I have one shortcut called Skullgirls.desktop, which is installed by steam:

[Desktop Entry]
Name=Skullgirls
Comment=Play this game on Steam
Exec=steam steam://rungameid/245170
Icon=steam_icon_245170
Terminal=false
Type=Application
Categories=Game;

Searching for that icon reveals:

~/.local/share/icons $ find . -iname \*steam_icon_245170\*

./hicolor/128x128/apps/steam_icon_245170.png
./hicolor/32x32/apps/steam_icon_245170.png
./hicolor/256x256/apps/steam_icon_245170.png
./hicolor/48x48/apps/steam_icon_245170.png
./hicolor/16x16/apps/steam_icon_245170.png

and this looks fully consistent with the icon theme spec linked above. However, the desktop doesn't render this icon, instead giving me what appears to be the default icon for a random document.

**Comment by palinek** [comment]
pcmanfm-qt (which is responsible for rendering desktop in lxqt) uses QIcon::fromTheme for getting the icons (which is aware of XDG_CONFIG_DATA http://doc.qt.io/qt-5/qicon.html#themeSearchPaths).

Is the XDG_CONFIG_DATA env variable set correctly? (just tested this with specific icon from *~/.local/share/icons* and icon is showed correctly)

**Comment by Jemi** [comment]
Update: I attempted to work around this problem by creating  ~/.local/share/icons/hicolor/index.theme:

---

[Icon Theme]
Name=hicolor
Comment=Fallback theme
Hidden=Yes
Directories=16x16/apps,16x32/apps,20x20/apps,24x24/apps,32x32/apps,32x64/apps,48x48/apps,64x64/apps,96x96/apps,128x128/apps,256x256/apps

[16x16/apps]
Size=16
Type=Fixed
Context=Applications

[16x32/apps]
Size=16
Type=Fixed
Context=Applications

[20x20/apps]
Size=20
Type=Fixed
Context=Applications

[24x24/apps]
Size=24
Type=Fixed
Context=Applications

[32x32/apps]
Size=32
Type=Fixed
Context=Applications

[32x64/apps]
Size=32
Type=Fixed
Context=Applications

[48x48/apps]
Size=48
Type=Fixed
Context=Applications

[64x64/apps]
Size=64
Type=Fixed
Context=Applications

[96x96/apps]
Size=96
Type=Fixed
Context=Applications

[128x128/apps]
Size=128
Type=Fixed
Context=Applications

[256x256/apps]
Size=256
Type=Fixed
Context=Applications

---

I don't know what I did wrong here because now my desktop is the contents of my system root directory [bin  dev  home  lib32  lost+found  opt etc ... ]. I renamed the index.theme file to index.theme.bak, but after logging out and logging back in, it's still the same. Anyway....


To answer your question, XDG_CONFIG_DATA is unset. I gather this is wrong, but I don't know if it became unset as a result of what I did, or if it was already unset. Please let me know what I should set it to (and how to get my desktop to show my desktop shortcuts, if  possible). Thanks.

**Comment by palinek** [comment]
The XDG_CONFIG_DATA was a typo, I've updated the comment.

**Comment by Jemi** [comment]
Yeah. It's set correctly.

**Comment by palinek** [comment]
> Yeah. It's set correctly.

What is the value?

Is it set before pcmanfm-qt is run?
You can:
- try to stop all instances of pcmanfm-qt (all shown pcmanfm-qt windows + desktop by lxqt-config-session)
- start *pcmanfm-qt --desktop* from terminal (before check if XDG_DATA_DIRS is set correctly)

**Comment by Jemi** [comment]
rogue ~ $ echo $XDG_DATA_DIRS
/usr/local/share:/usr/share:/home/rogue/.local/share

This is before starting pcmanfm-qt --desktop. Subsequently to running that command, my desktop comes up but without any background, and again showing the system root instead of ${HOME}/Desktop.

**Comment by Jemi** [comment]
I've been able to get my desktop shortcuts back by deleting all configuration files related to pcmanfm-qt or lxqt. Subsequently to that, I went to the LXQt Appearance Configuration screen and looked at icon themes. It detected the Flattr icon theme that I've installed to ~/.icons. This theme has a terminal.svg icon which pcmanfm-qt will load if it is selected in a .desktop file by absolute path, e.g.:

[Desktop Entry]
Exec=urxvt -bg black -fg white
Icon=/home/rogue/.icons/flattr-icons/apps/scalable/terminal.svg
Terminal=false
Type=Application

Replacing the full icon path with just terminal, e.g.:

Icon=terminal

from my understanding of the spec, *should* work , but it does not, and displays no icon at all. Trying with the other themes detected by LXQt Appearance configuration, I find the same thing. Most of these themes are installed system wide to /usr/share/icons. But even though most if not all of these themes have an icon matching the pattern terminal.*, none of these icons display for my urxvt shortcut:

[Desktop Entry]
Exec=urxvt -bg black -fg white
Icon=terminal
Terminal=false
Type=Application

Based on this, it's clear to me that pcmanfm-qt, at least in its 0.10.0 version, implements basically none of the Icon Theme Specification.

**Comment by Jemi** [comment]
Found a workaround:

cd ~/.icons
ln -s ../.local/share/icons/hicolor hicolor

this allows pcmanfm-qt to see application installed icons.

**Comment by dglent** [comment]
As says here the icons should be located in the hicolor path
http://standards.freedesktop.org/icon-theme-spec/icon-theme-spec-latest.html#install_icons

**Comment by Jemi** [comment]
Yes, it's supposed to go in  $prefix/share/icons/hicolor/48x48/apps. But $prefix, if I'm not mistaken, refers to some writeable directory in $XDG_DATA_DIRS. ${HOME}/.icons should not be in that path according to the spec:

http://standards.freedesktop.org/icon-theme-spec/icon-theme-spec-latest.html#directory_layout

${HOME}/.icons is included in the search path of programs needing to load icons only for backwards compatibility. Programs needing to *place* an icon somewhere need to look in $XDG_DATA_DIRS.

**Comment by pmattern** [comment]
*usage of root directory as desktop directory*
A potential reason is tweaking of those directories defined by [xdg-user-dirs](http://www.freedesktop.org/wiki/Software/xdg-user-dirs) from "LXQt Session Settings" (binary `lxqt-config-session`) while binaries `xdg-user-dir[s-update]` are not available, see #896. This in turn may take place if the package providing lxqt-session does not depend on the one providing binaries `xdg-user-dir[s-update]` which is not that unlikely right now as the option to tweak those directories got introduced [only recently](https://github.com/lxde/lxqt/issues/780).

*actual problem*
Basically this is caused by a Qt bug and does not affect the desktop only but context menus to pick applications within the actual file manager as well. A possible workaround exists but hasn't been applied yet. See lxde/pcmanfm-qt#198.
That said I cannot reproduce your findings with some dummy file `juff_ed_123456.png`. It doesn't get displayed if placed in `/usr/share/pixmaps/` or `~/.[local/share/]icons` but does get displayed in `~/.[local/share/]icons/hicolor/48x48/apps/` (as well as `/usr/share/icons/hicolor/48x48/apps/`). Latest VCS checkouts of all LXQt components on Arch Linux, `$XDG_DATA_DIRS` set to `/home/<user>/.local/share:/usr/local/share:/usr/share`. So maybe something got changed in the code to cover `~/.local/share/` recently?
That aside this is issue number four addressing the problem. Maybe this discussion as a whole or at least its results as far as `~/.local/share` is concerned should go into lxde/pcmanfm-qt#198 so as to avoid further clutter.
